sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -500,53 +499,73 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
      * @param offset Optional starting point of the upload range. It will start from the beginning if it is {@code
      * null}
-     * @param length Specifies the number of bytes being transmitted in the request body. When the FileRangeWriteType is
-     * set to clear, the value of this header must be set to zero.
-     * @param type You may specify one of the following options:
-     * <ul>
-     * <li>Update: Writes the bytes specified by the request body into the specified range.</li>
-     * <li>Clear: Clears the specified range and releases the space used in storage for that range. To clear a range,
-     * set the Content-Length header to zero.</li>
-     * </ul>
      * @return A response that only contains headers and response status code
      * @throws StorageErrorException If you attempt to upload a range that is larger than 4 MB, the service returns
      * status code 413 (Request Entity Too Large)
      */
-    public Mono<Response<FileUploadInfo>> upload(Flux<ByteBuffer> data, long length, long offset, FileRangeWriteType type) {
+    public Mono<Response<FileUploadInfo>> upload(Flux<ByteBuffer> data, long length, long offset) {
         FileRange range = new FileRange(offset, offset + length - 1);
-        return azureFileStorageClient.files().uploadRangeWithRestResponseAsync(shareName, filePath, range.toString(), type, length, data, null, null, Context.NONE)
+        return azureFileStorageClient.files().uploadRangeWithRestResponseAsync(shareName, filePath, range.toString(), FileRangeWriteType.UPDATE, length, data, null, null, Context.NONE)
             .map(this::uploadResponse);
     }
 
     /**
-     * Uploads file to storage file service.
+     * Clear a range of bytes to specific of a file in storage file service. Clear operations performs an in-place
+     * write on the specified file.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * <p>Upload the file from the source file path. </p>
+     * <p>Clears the first 1024 bytes. </p>
      *
-     * {@codesnippet com.azure.storage.file.fileAsyncClient.uploadFromFile#string}
+     * {@codesnippet com.azure.storage.file.fileAsyncClient.clearRange#long}
      *
      * <p>For more information, see the
-     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-file">Azure Docs Create File</a>
-     * and
-     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/put-range">Azure Docs Upload</a>.</p>
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/put-range">Azure Docs</a>.</p>
      *
-     * @param uploadFilePath The path where store the source file to upload
-     * @return An empty response.
+     * @param length Specifies the number of bytes being cleared in the request body.
+     * @return A response that only contains headers and response status code
+     * @throws StorageErrorException If you attempt to clear a range that is larger than 4 MB, the service returns
```

Comment:
Remove the statement about 4MB, clear is allowed a size up to the size of the file.

Reply:
Done.